rosidl_typesupport_cpp/test/test_message_type_support_dispatch.cpp

```diff
@@ -124,3 +125,24 @@ TEST(TestMessageTypeSupportDispatch, get_handle_function) {
       &type_support_cpp_identifier,
       "test_type_support4"), nullptr);
 }
+
+TEST(TestMessageTypeSupportDispatch, maybe_fail_test)
+{
+  rosidl_message_type_support_t type_support_cpp_identifier =
+    get_rosidl_message_type_support(rosidl_typesupport_cpp::typesupport_identifier);
+  rcpputils::SharedLibrary * library_array[map_size] = {nullptr, nullptr, nullptr};
+  type_support_map_t support_map = get_typesupport_map(reinterpret_cast<void **>(&library_array));
+  type_support_cpp_identifier.data = &support_map;
+
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    // load library and find symbols
+    try {
+      auto * result = rosidl_typesupport_cpp::get_message_typesupport_handle_function(
+        &type_support_cpp_identifier,
+        "test_type_support1");
+      EXPECT_NE(result, nullptr);
+    } catch (...) {
```

Comment:
Why the `try` `catch` block in this scenario? if it is an expected throw, maybe `EXPECT_THROW` can be used? (Not clear to me if possible though)

Reply:
Because it doesn't throw for all potential errors. In some instances it will just return a `nullptr`.

Reply:
Should we be catching specific exceptions? Or in other words, doesn't the API specify what exceptions it may throw?

Reply:
I don't think the API does say anything about it, but adding a specific catch for the runtime errors seems reasonable.

Reply:
Added (btw)

Reply:
At some point, we should document potential exceptions just like we document error codes.